WalletWasabi.Daemon/Config.cs

```diff
@@ -0,0 +1,186 @@
+	public string[] JsonRpcServerPrefixes => GetEffectiveValue(PersistentConfig.JsonRpcServerPrefixes, x => new [] { x });
+	public Money DustThreshold => GetEffectiveValue(PersistentConfig.DustThreshold, x =>
+	{
+		if (Money.TryParse(x, out var money))
+		{
+			return money;
+		}
+		throw new ArgumentNullException("DustThreshold", "Not a valid money");
+	});
+
+	public bool BlockOnlyMode => GetEffectiveBool(false, "blockonly");
+	public string LogLevel => GetEffectiveString("", "loglevel");
+
+	public static string DataDir => GetString(
+		EnvironmentHelpers.GetDataDir(Path.Combine("WalletWasabi", "Client")),
+		Environment.GetCommandLineArgs(),
+		"datadir");
+
+	public bool EnableGpu => GetEffectiveBool(PersistentConfig.EnableGpu);
+	public string CoordinatorIdentifier => GetEffectiveString(PersistentConfig.CoordinatorIdentifier);
+	public ServiceConfiguration ServiceConfiguration => new (GetBitcoinP2pEndPoint(), DustThreshold);
+
+	private EndPoint GetEffectiveEndPoint(
+		EndPoint valueInConfigFile,
+		[CallerArgumentExpression("valueInConfigFile")] string key = "") =>
+		GetEffectiveValue(
+			valueInConfigFile,
+			x =>
+			{
+				if (EndPointParser.TryParse(x, 0, out var endpoint))
+				{
+					return endpoint;
+				}
+
+				throw new ArgumentNullException(key, "Not a valid endpoint");
+			},
+			Args,
+			key);
+
+	private bool GetEffectiveBool(
+		bool valueInConfigFile,
+		[CallerArgumentExpression("valueInConfigFile")] string key = "") =>
+		GetEffectiveValue(
+			valueInConfigFile,
+			x => x switch
+			{
+				_ when x.Equals("true", StringComparison.InvariantCultureIgnoreCase) => true,
+				_ when x.Equals("false", StringComparison.InvariantCultureIgnoreCase) => false,
+				_ => throw new ArgumentNullException(key, "must be 'true' or 'false'")
+			},
+			Args,
+			key);
+
+	private string GetEffectiveString(
+		string valueInConfigFile,
+		[CallerArgumentExpression("valueInConfigFile")] string key = "") =>
+		GetEffectiveValue(valueInConfigFile, x => x, Args, key);
+
+	private string? GetEffectiveOptionalString(
+		string? valueInConfigFile,
+		[CallerArgumentExpression("valueInConfigFile")] string key = "") =>
+		GetEffectiveValue(valueInConfigFile, x => x, Args, key);
+
+	private T GetEffectiveValue<T>(T valueInConfigFile, Func<string, T> converter, [CallerArgumentExpression("valueInConfigFile")] string key = "") =>
+		GetEffectiveValue(valueInConfigFile, converter, Args, key);
+
+	private static string GetString(
+		string valueInConfigFile, string[] args, [CallerArgumentExpression("valueInConfigFile")] string key = "") =>
+		GetEffectiveValue(valueInConfigFile, x => x, args, key);
+
+	private static T GetEffectiveValue<T>(T valueInConfigFile, Func<string, T> converter, string[] args, [CallerArgumentExpression("valueInConfigFile")] string key = "")
+	{
+		key = key.StartsWith(nameof(PersistentConfig) + ".")
+			? key.Remove(0, nameof(PersistentConfig).Length + 1)
+			: key;
+
+		var cliArgKey = ("--" + key + "=");
```

Comment:
I don't like this because it uses the Culture. I prefer to concat.

Reply:
Not so, the proposal by @kiminuo is exactly the same as you wrote, but more laconic (see it at [SharpLab](https://sharplab.io/#v2:D4AQTAjAsAUCDMACciDCiDetE+UkEAbMhAAyIBqEAFAeQNYCmAngJSIC8AfIgCQBEAWkEYmzAL4d+Abmy45OBCWJ1KYWmURj23RNSGD+iANRaWJxPymtZMcUA===)).

Reply:
Yup, interpolation is preferred now days because the compiler can decide the most optimal path for them depending on the use case: concatenation, creation of a formatted string or an interpolation.